src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -451,30 +451,40 @@ public FormValidation doCheckProjectKey(@QueryParameter String value) throws IOE
             return FormValidation.ok();
         }
 
-        public ListBoxModel doFillPriorityIdItems() {
+        public ListBoxModel doFillPriorityIdItems(@AncestorInPath Item item) {
+            Project project = null;
             ListBoxModel items = new ListBoxModel().add(""); // optional field
-            for (JiraSite site : JiraGlobalConfiguration.get().getSites()) {
-                JiraSession session = site.getSession();
-                if (session != null) {
-                    for (Priority priority : session.getPriorities()) {
-                        items.add("[" + site.getName() + "] " + priority.getName(), String.valueOf(priority.getId()));
+            if(item instanceof Project) {
+                project = (Project) item;
+                final JiraProjectProperty jiraProjectProperty = (JiraProjectProperty) project.getProperty(JiraProjectProperty.class);
+                JiraSite site = jiraProjectProperty.getSite();
+                if(null != site) {
+                    JiraSession session = site.getSession();
+                    if (session != null) {
+                        for (Priority priority : session.getPriorities()) {
+                            items.add("[" + site.getName() + "] " + priority.getName(), String.valueOf(priority.getId()));
+                        }
                     }
                 }
             }
             return items;
         }
 
-        public ListBoxModel doFillTypeIdItems() {
+        public ListBoxModel doFillTypeIdItems(@AncestorInPath Item item) {
+            Project project = null;
             ListBoxModel items = new ListBoxModel().add(""); // optional field
-            for (JiraSite site : JiraGlobalConfiguration.get().getSites()) {
+            if(item instanceof Project) {
+                project = (Project) item;
+                final JiraProjectProperty jiraProjectProperty = (JiraProjectProperty) project.getProperty(JiraProjectProperty.class);
+                JiraSite site = jiraProjectProperty.getSite();
                 JiraSession session = site.getSession();
                 if (session != null) {
                     for (IssueType type : session.getIssueTypes()) {
                         items.add("[" + site.getName() + "] " + type.getName(), String.valueOf(type.getId()));
```

Comment:
Do you think sorting the list alphabetically makes sense? (or putting the globally defined sites at the top?)

Reply:
Maybe I misunderstood this part, but for me the list of issues types and priorities are got from the Jira API. It must use the connexion to Jira configured on the Project properties and not in Global Configuration.